gui/elements/qt/_viewer.py

```diff
@@ -17,11 +17,15 @@ def __init__(self, viewer):
         layout = QGridLayout()
         self.setLayout(layout)
         layout.setContentsMargins(0, 0, 0, 0)
-        layout.setColumnStretch(0, 4)
+        #layout.setColumnStretch(0, 4)
 
         row = 0
         layout.addWidget(self.canvas.native, row, 0)
-        layout.setRowStretch(row, 1)
+        #layout.setRowStretch(row, 1)
+
+        self.layersLayout = QVBoxLayout()
```

Comment:
I like this idea a lot and am currently working on its implementation but one thing is unclear to me. Do I want to do something like add a `controls`, `viewer`, and `layerList` all to the main `window` or do I want to keep the `window` unchanged and add the `controls` and `layerList` to the `viewer` and then add the `viewer` to the window. Or maybe I want to keep `viewer` and `window` as it was and add some intermediate layout which is the `HBoxLayout` that would contain a `controls`, `viewer`, and `layerList` and then add that to the `window`.

The first one seems cleaner to me, aside from the idea that the `window` could contain multiple `viewers`, not quite sure how to think about that in that case. Are we supporting adding multiple viewers to the same window, would they appear as tabs? In that case the last option seems to be the best. Any guidance would be appreciated.

Reply:
I've given this a try and pushed the new commits. I've sort of done something most like the first / second options combined - when you `add_viewer` to a `window` it will go through and add a `controls` - currently two vertical sliders not linked up to anything (but can become the clims and gamma sliders), then it adds a `viewer` and then it adds a `layersList`